### ISSUE TITLE
French d-portal doesn’t use French codelist values

### ISSUE THREAD

**Issue by andylolz** [devinit/D-Portal#452]
This looks non-trivial to fix, but I thought I’d raise it anyway.

The (3-digit) sectors listed e.g. here:
http://d-portal.org/ctrack.html?search=&publisher=46002&tongue=fra#view=sectors

…are mostly available in French here:
http://iatistandard.org/203/codelists/downloads/clv2/xml/SectorCategory.xml
(and NB [5-digit sector codes are also available](http://iatistandard.org/203/codelists/downloads/clv2/xml/SectorCategory.xml))

…but English is displayed. Is it possible to use French codelist values, where available?

**Comment by xriss** [comment]
yup.

Stage 1.

Either we need to switch from the json to the xml as source or IATI needs to publish better json.

Stage 2.

We pickup the requested language and then merge what data we have for that language with the English version as backup when codes are missing.


Personally I'd quite like stage 1 to happen at iatistandard.org before we start using it, it  would be nice to have more than just French/English and even in the English version SOME OF THE CODES ARE IN ALL CAPS for no apparent reason so a general cleanup of all codelists and publishing multiple languages in the json and csv not just the xml would be a good start.


PS: Any idea what the clv1 clv2 clv3 is all about? I just pick the bigger number on the grounds that bigger numbers are probably better...

**Comment by andylolz** [comment]
> Either we need to switch from the json to the xml as source or IATI needs to publish better json.

Well, so I don’t know if it helps, but the French translations are available in a separate JSON:
http://iatistandard.org/203/codelists/downloads/clv3/json/fr/SectorCategory.json

> Personally I'd quite like stage 1 to happen at iatistandard.org before we start using it, it would be nice to have more than just French/English

@yohannaloucheur is also [keen for this to happen](https://discuss.iatistandard.org/t/show-iati-multilingual-iati-standard-site/1266/5)!

> SOME OF THE CODES ARE IN ALL CAPS

Haha, yep… So that’s an issue with the source DAC codelist, but I agree that that could reasonably be cleaned up in the IATI data. I’ll send a pull request for that.

> PS: Any idea what the clv1 clv2 clv3 is all about? I just pick the bigger number on the grounds that bigger numbers are probably better...

clv2 and clv3 both support multiple languages, but clv3 does it in the more IATI way (i.e. with narrative elements). So your approach is totally correct :)

**Comment by notshi** [comment]
#394

**Comment by notshi** [comment]
Thanks, @andylolz - also raised in #394

**Comment by andylolz** [comment]
By me?! Ha, sorry! I totally forgot about that.

**Comment by YohannaLoucheur** [comment]
Yeah, glad to see this happening!

Re clv 1, 2 and 3, I may not understand fully how these things work, but don't we have to use the relevant version depending on the standard version used by the publisher?  ie if a publisher is still using v1.03, then the data won't display properly with clv 2 or 3?  And clv 3 would only work for publishers using 2.0*?

**Comment by andylolz** [comment]
It’s true that clv3 is only available for IATI-Codelists v2.0x. Though I don’t think there’s any technical reason for this. I raised this as an issue recently, in fact: IATI/IATI-Codelists#153

**Comment by notshi** [comment]
Closed as blocked.

**Comment by andylolz** [comment]
Can’t remember if I have mentioned elsewhere / don’t know if it helps clear the blockage, but https://codelists.codeforiati.org exists and might be useful here.
